libs/ballot-interpreter/src/adjudication_reasons.test.ts

```diff
@@ -299,3 +303,116 @@ test('an unmarked write-in can trigger the overvote reason', () => {
+      lion: { markStatus: MarkStatus.Marked },
+      kangaroo: { markStatus: MarkStatus.Marked },
+    }
+  );
+
+  const mockContestOptionScoresWithoutZebra = mockContestOptionScores.filter(
+    (c) => c.option.id !== 'zebra'
+  );
+  const markedZebraOption = find(
+    mockContestOptionScores,
+    (c) => c.option.id === 'zebra'
+  );
+  const marginalZebraOption: GetAllPossibleAdjudicationReasonsOptionStatus = {
+    ...markedZebraOption,
+    markStatus: MarkStatus.Marginal,
+  };
+  const unmarkedZebraOption: GetAllPossibleAdjudicationReasonsOptionStatus = {
+    ...markedZebraOption,
+    markStatus: MarkStatus.Unmarked,
+  };
+
+  const marginalAdjudicationReason: AdjudicationReasonInfo = {
+    type: AdjudicationReason.MarginalMark,
+    contestId: zooCouncilMammal.id,
+    optionId: zooCouncilMammalCandidate1.id,
+    optionIndex: 0,
+  };
+  const undervoteAdjudicationReason: AdjudicationReasonInfo = {
+    type: AdjudicationReason.Undervote,
+    contestId: zooCouncilMammal.id,
+    optionIds: ['lion', 'kangaroo'],
+    optionIndexes: [1, 2],
+    expected: 3,
+  };
+
+  test.each<{
+    description: string;
+    zebraOptionA: GetAllPossibleAdjudicationReasonsOptionStatus;
+    zebraOptionB: GetAllPossibleAdjudicationReasonsOptionStatus;
+    expected: AdjudicationReasonInfo[];
+  }>([
+    {
+      description: 'marked, marked',
+      zebraOptionA: markedZebraOption,
+      zebraOptionB: markedZebraOption,
+      expected: [],
+    },
+    {
+      description: 'marked, unmarked',
+      zebraOptionA: markedZebraOption,
+      zebraOptionB: unmarkedZebraOption,
+      expected: [],
+    },
+    {
+      description: 'unmarked, marked',
+      zebraOptionA: unmarkedZebraOption,
+      zebraOptionB: markedZebraOption,
+      expected: [],
+    },
+    {
+      description: 'marked, marginal',
+      zebraOptionA: markedZebraOption,
+      zebraOptionB: marginalZebraOption,
+      expected: [],
+    },
+    {
+      description: 'marginal, marked',
+      zebraOptionA: marginalZebraOption,
+      zebraOptionB: markedZebraOption,
+      expected: [],
+    },
+    {
+      description: 'marginal, marginal',
+      zebraOptionA: marginalZebraOption,
+      zebraOptionB: marginalZebraOption,
+      expected: [marginalAdjudicationReason, undervoteAdjudicationReason],
+    },
+    {
+      description: 'marginal, unmarked',
+      zebraOptionA: marginalZebraOption,
+      zebraOptionB: unmarkedZebraOption,
+      expected: [marginalAdjudicationReason, undervoteAdjudicationReason],
+    },
+    {
+      description: 'unmarked, marginal',
+      zebraOptionA: unmarkedZebraOption,
+      zebraOptionB: marginalZebraOption,
+      expected: [marginalAdjudicationReason, undervoteAdjudicationReason],
+    },
+    {
+      description: 'unmarked, unmarked',
+      zebraOptionA: unmarkedZebraOption,
+      zebraOptionB: unmarkedZebraOption,
+      expected: [undervoteAdjudicationReason],
+    },
+  ])(
```

Comment:
Would be nice to add some kind of test for overvote detection, too. I see that that might require a different contest, e.g, a vote for 1